source/blog/2018-07-10-Skydive-With-oVirt.html.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Skydive With oVirt 
+author: mburman
+tags: oVirt, oVirt 4.2, open source virtualization,network
+date: 2018-07-10 09:01:00 UTC
+---
+
+## Summary
+
+Skydive is an open source real-time network topology and protocols analyzer providing a comprehensive way of understanding what is happening in your network infrastructure.
+The common use cases will be, troubleshooting, monitoring, SDN integration and much more.
+It has features such as:
+* Topology capturing - Captures network topology, interface, bridge and more
+* Flow capture - Distributed probe, L2-L4 classifier, GRE, VXLAN, GENEVE, MPLS/GRE, MPLS/UDP tunnelling support
+* Extendable - Support for external SDN Controllers or container based infrastructure, OpenStack, OpenContrail, Docker. Supports extensions through API
+* Distributed
+
+[Skydive network](http://skydive.network/) - Link to Skydive site
+
+## Installation steps
+1. git clone https://github.com/skydive-project/skydive.git
+2. Create inventory file
+
+```ini  
+[skydive:children]
+analyzers
+agents
+
+[skydive:vars]
+skydive_listen_ip=0.0.0.0
+skydive_fabric_default_interface=ovirtmgmt
+
+skydive_os_auth_url=https://<FQDN>:35357/v2.0
+skydive_os_service_username=<username>
+skydive_os_service_password=<password>
+skydive_os_service_tenant_name=service
+skydive_os_service_domain_name=Default
+skydive_os_service_region_name=RegionOne
+
+[analyzers]
+<FQDN> ansible_ssh_user=root ansible_ssh_pass=<password>
+
+[agents]
+<FQDN> ansible_ssh_user=root ansible_ssh_pass=<password>
+<FQDN> ansible_ssh_user=root ansible_ssh_pass=<password>
+<FQDN> ansible_ssh_user=root ansible_ssh_pass=<password>
+<FQDN> ansible_ssh_user=root ansible_ssh_pass=<password>
+
+[agents:vars]
+skydive_extra_config={'agent.topology.probes': ['ovsdb', 'neutron'], 'agent.topology.neutron.ssl_insecure': true}
+```
+* skydive_os_auth_url - This is the FQDN(hostname or IP) address of ovirt-provider-ovn
+* skydive_os_service_username - oVirt username used to authenticate the ovirt-provider-ovn, e.g. admin@internal
+* analyzer FQDN will be the hostname of your analyzer
+* agent FQDN will be the hostname of the hosts running in oVirt
+
+3. cd to git/skydive/contrib/ansible
+4. Run ansible-playbook -i <inventory file> playbook.yml.sample
+5. Connect to skydive UI http://<IP/FQDN>:8082
+6. If the UI cannot be accessed flush the iptables 
+  
+## Benefit to oVirt users
+oVirt users may now analyze and see thier network topology via the Skydive UI. They may capture traffic from VM1 to VM2 for example or monitor the traffic between two VMs or hosts. 
+Also they may generate some traffic between 2 running VMs on different hosts for example.
+Users may create alerts in Skydive UI to notify that some traffic disconnected or down.
+
```

Comment:
oVirt is good to create configuration for hosts and VMs. Skydive enables the users to see the result of the configuration, especially how network interfaces of hosts are connected, e.g.
- Analyse how the NICs of hosts are plugged and how broad cast domins (VLANs) are configured in the switch. This might be helpful for the initial configuration of oVirt.

- The VLAN with tag 4 is connected to second NIC in host 1 and second NIC in host 2, while the second NIC of host 0 is not connected to the same broad cast domain, even if the second NIC of host 0 is configured to use VLAN with tag 4, too. This might happen if the switch is configured in an unexpected way or a network cable is plugged in an unexpected way. This might be helpful for manual interactive error analysis.

- To detect which logical network is used to transport the tunneled OVN data.

Is there already a way to show the connection of the VMs?

Reply:
Hmm, you don't have the logical network which is used for the tunnelled OVN. 
We still have some issues to see the neutron networks IDs.
At the moment you can see which vnet is connected to the br-int, but you have no idea which network is uses or which ovn port. 
I will add example and screenshot when we will figure this out. 
Cheers)

Reply:
To detect which logical network is used to transport the tunneled OVN data, there could be a capture of the traffic if the physical network which shows the tunneling and the tunneled data. I am going to send you an example pcap via email. Such a capture could be embedded in an explained complete scenario, e.g. including the MACs of the VMs and the MAC and IP addresses of the hosts.

Reply:
Dominik, thank you for the comments.
I really don't want to start explaining the skydive features. Using capture and generate traffic in skydive is part of skydive. This is something that the admins will have to learn by they own.
I will give 2 examples with short description. 
This is fine by me and dan. If you feel that you want to add complete explained scenarios, be my guest. From my point of view this is good enough)